pkg/pcap/pcaps.go

```diff
@@ -0,0 +1,95 @@
+package pcap
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/aquasecurity/tracee/pkg/events"
+	"github.com/aquasecurity/tracee/pkg/logger"
+	"github.com/aquasecurity/tracee/pkg/utils"
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+//
+// This is a big Pcaps struct holding caches for different types of Pcap files
+// to be managed (pcap files per process, per containers and per commands). It
+// would be hard to keep all possible pcap files open forever (as tracee might
+// trace tons of processes, containers or commands).
+//
+// That is what the PcapCache struct is for. It keeps Pcap files and maintain
+// those Pcap files cached while they're still being used.
+//
+// At the end we have the Pcap struct itself. It describes a pcap file being
+// kept opened on behalf of a process, a container or a command.
+//
+// NOTE: Pcaps is not thread safe, should be called from a single routine.
+// TODO: might replace cache_XXX and pcap_XXX files with generics.
+//
+
+// Pcaps holds all Pcap for different PcapTypes
+type Pcaps struct {
+	pcapCaches map[PcapType]*PcapCache
+}
+
+func NewPcaps(config PcapType, output *os.File) (*Pcaps, error) {
```

Comment:
I'm going to stress again that this should be a regular struct config.
I'll lay out my argument again:
1. This is the exception in the project
2. It's not as intuitive for use later in debugging and reading internally.
3. More importantly, it is externally harder to use, especially for non cli users who will have to configure this by hand. Letting them do bit flags to configure is not ok from a consumer standpoint.
4. In addition since we have other configuration options, like the file, which you allow configuring in this function, it shows that it should be open to non-cli users too. It would be easiest if a struct was parsed for the output directory location.
5. If we ever want to add new configuration options, rather than breaking the function signature we simply add a new **non-breaking** struct field.
I think these are enough important point to move from the flag approach. If it's that important, maybe convert it to a  bit flag internally for use, but at least externally, it should be simple bool flags, with the other configuration options (for now the output dir, in the future potentially more).

Reply:
Alright, the external user reason (using it as a lib) convinced me.